profiles/targets/clang/make.defaults

```diff
@@ -0,0 +1,6 @@
+# Copyright 1999-2016 Gentoo Foundation.
+# Distributed under the terms of the GNU General Public License, v2
+# $Id$
+
+CC="clang"
+CXX="clang++"
```

Comment:
I think that it is better to add BUILD_CC="${CC}" and BUILD_CXX="${CXX}" to make.defaults.
It will make clang work more, in most cases gcc will be unnecessary.

Please see bug 595878, amd64-fbsd/clang profile: pure clang support.
https://bugs.gentoo.org/show_bug.cgi?id=595878

I have never tried clang on Gentoo/Linux, so would you test it ?

Test steps)

CHOST="x86_64-pc-linux-gnu" (Of course please enter your CHOST.)
for f in c++ cc cpp g++ gcc gfortran
do
[[ -e /usr/bin/${f} ]] && rm -v /usr/bin/${f}
[[ -e /usr/bin/${CHOST}-${f} ]] && rm -v /usr/bin/${CHOST}-${f}
done
mv /usr/${CHOST}/gcc-bin /usr/${CHOST}/gcc-bin.orig

emerge '=sys-libs/ncurses-5.9-r5'
(Probably, you get an error message that gcc does not exist.)

Add BUILD_CC="${CC}" and BUILD_CXX="${CXX}" to profiles/targets/clang/make.defaults
And retry...
emerge '=sys-libs/ncurses-5.9-r5'


Restore)
mv /usr/${CHOST}/gcc-bin.orig /usr/${CHOST}/gcc-bin
unset CHOST
gcc-config 1